var/ramble/repos/builtin/applications/fio/application.py

```diff
@@ -0,0 +1,584 @@
+# Copyright 2022-2025 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import os
+
+from ramble.appkit import *
+
+from spack.util.path import canonicalize_path
+
+
+class Fio(ExecutableApplication):
+    """Flexible I/O Tester. Fio spawns a number of threads or processes doing a
+    particular type of I/O action as specified by the user. fio takes a number
+    of global parameters, each inherited by the thread unless otherwise
+    parameters given to them overriding that setting is given.
+    """
+
+    name = "fio"
+
+    maintainers("dapomeroy")
+
+    tags("io-benchmark", "storage-benchmark")
+
+    with when("package_manager_family=spack"):
+        define_compiler("gcc13", pkg_spec="gcc@13.1.0")
+
+        software_spec("fio", pkg_spec="fio@3.37 +libaio")
+
+    executable(
+        "run",
+        template=[
+            "echo 'Running job file {job_file}...'",
+            "fio --output={out_file} --output-format {out_format} --eta=never {job_file_path}",
+        ],
+        use_mpi=False,
+    )
+
+    # Use client/server mode to run on multiple nodes
+    executable(
+        "slurm-multinode-run",
+        template=[
+            "export DIRECTORY='{directory}'",
+            "echo 'Starting fio servers...'",
+            "pdsh -w $SLURM_JOB_NODELIST {experiment_run_dir}/fio_start_server.sh",
```

Comment:
Can change `$SLURM_JOB_NODELIST` to `{hostlist}` and make this a variable you have to define.

Workflow managers (like slurm) will then define this for you so you don't need to define it always.